_posts/2021-10-09-software-and-projects-as-systems.md

```diff
@@ -0,0 +1,191 @@
+---
+layout: post
+title: Software Engineering and everything related - Software and Project as systems
+author: tommi.kenakkala
+excerpt: Have you ever wondered why a change to software produces unexpected results? Taking a system theory flavored view might help software developers. Read on to find out more!
+tags:
+  - Software engineering
+  - Project management
+  - Ways of working
+  - System Theory
+---
+
+## Foreword
+
+Have you ever wondered why a change to software produces unexpected results?
+
+Software designers build systems.
+
+Taking a system theory flavored view might help. Following may help fresh software developers especially on seeing the big picture around them in software projects.
+
+Read on to find out more!
+
+## System
+
+[System](https://en.wikipedia.org/wiki/System) is a set of things working together.
+
+- Systems are composed of other systems
+  - Examples
+    - a bridge
+    - the Solar System (it's in the name after all!)
+    - software, its operating environment and people operating it
+- Software systems are produced by other systems
+  - Examples
+    - software project
+    - team
+    - organization
+
+## Change
+
+A change is something which alters the structure or state of a **system**.
+
+A change may take a system a step towards a desired state or away from it.
+
+One can argue that the whole **art of software engineering simply produces changes to systems**.
+
+There are no features, bugs, change requests and whatnot. They are all equally and simply changes to a specific system. The change in the behavior or state of that system in turn affects other systems.
```

Comment:
"There are no features, bugs, change requests and whatnot" koska lauseen alussa on kielto, minun kielikorvani sanoisi että tuon andin pitäisi olla or. Mutta en ole satavarma.

Reply:
Roger!